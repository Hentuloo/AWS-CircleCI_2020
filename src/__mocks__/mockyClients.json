{
  "data": [
    {
      "address": {
        "city": "Warsaw",
        "street": "Miodowa",
        "houseNumber": "18"
      },
      "_id": "5e2ff7e09b56431b7865ef2d",
      "name": "Lewandowski Andrue",
      "email": "andrueLew@o2.pl",
      "sex": false,
      "avatar": "https://cdn.pixabay.com/photo/2013/07/13/10/07/man-156584_960_720.png",
      "age": 22,
      "__v": 0
    },
    {
      "address": {
        "city": "Gerenada",
        "street": "call de christo",
        "houseNumber": "23A"
      },
      "_id": "5e2ff81a9b56431b7865ef2e",
      "name": "Julian Ochota",
      "email": "julian@fmail.com",
      "sex": true,
      "avatar": "https://cdn4.iconfinder.com/data/icons/professional-avatar-5/48/Public_relation_avatar_women_woman_profile_character_professions_2-512.png",
      "age": 33,
      "__v": 0
    }
  ]
}
