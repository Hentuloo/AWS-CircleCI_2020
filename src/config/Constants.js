export const Constants = {
  pl: {
    STATEMENTS: {
      buttonMoreInfoTitle: 'Więcej informacji',
      buttonMoreInfoAlt: 'Przycisk więcej informacji',
      findClient: 'Znajdź klienta...',
      sort: {
        alt: 'Sortowanie',
        email: 'Sortuj według alfabetu (email)',
        age: 'Sortuj według wieku',
        descending: 'malejąco',
        ascending: 'rosnąco',
      },
      sex: {
        alt: 'Płeć',
        male: 'Mężczyzna',
        female: 'Kobieta',
      },
      icons: {
        place: 'Miejsce zamieszkania',
        personIcon: 'avatar',
      },
    },
  },
};
