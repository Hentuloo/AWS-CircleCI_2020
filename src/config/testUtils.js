import React from 'react';
import PropTypes from 'prop-types';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import GlobalStyles from 'themes/GlobalStyles';
import theme from 'themes/mainTheme';

export const withTheme = (ui, { customTheme } = {}) => {
  const Wrapper = ({ children }) => (
    <ThemeProvider theme={customTheme || theme}>
      <GlobalStyles />
      <div>{children}</div>
    </ThemeProvider>
  );
  Wrapper.propTypes = {
    children: PropTypes.node.isRequired,
  };
  return {
    ...render(ui, { wrapper: Wrapper }),
  };
};
