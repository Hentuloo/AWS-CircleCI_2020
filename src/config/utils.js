export const sortByAlphabet = (array, parametr, reverse) => {
  if (reverse) {
    return array.sort((a, b) =>
      a[parametr].localeCompare(b[parametr]),
    );
  }
  return array.sort((a, b) => b[parametr].localeCompare(a[parametr]));
};

export const sortByNumber = (array, parametr, reverse) => {
  if (reverse) {
    return array.sort(
      (a, b) => parseFloat(b[parametr]) - parseFloat(a[parametr]),
    );
  }
  return array.sort(
    (a, b) => parseFloat(a[parametr]) - parseFloat(b[parametr]),
  );
};

export const isIncludeText = (text, filter) =>
  text.toLowerCase().includes(filter.toLowerCase());
