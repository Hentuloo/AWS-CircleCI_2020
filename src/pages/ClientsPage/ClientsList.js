import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import UserCard from 'components/UserCard';

const ClientsWrapper = styled.ul`
  ${({ theme }) => theme.mediaQuery.lg} {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
`;
const StyledUserCard = styled(UserCard)`
  margin: 25px auto;
`;

const Clients = ({ data }) => {
  if (data) {
    return (
      <ClientsWrapper>
        {data.map(
          ({
            address: { city, street, houseNumber },
            _id,
            name,
            email,
            sex,
            avatar,
            age,
          }) => (
            <StyledUserCard
              key={_id}
              $as="li"
              name={name}
              age={age}
              email={email}
              address={{
                city,
                street,
                houseNumber,
              }}
              sex={sex}
              avatar={avatar}
            />
          ),
        )}
      </ClientsWrapper>
    );
  }
  return null;
};

Clients.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      age: PropTypes.number.isRequired,
      email: PropTypes.string.isRequired,
      sex: PropTypes.bool.isRequired,
      avatar: PropTypes.string.isRequired,
      address: PropTypes.shape({
        city: PropTypes.string,
        street: PropTypes.string,
        houseNumber: PropTypes.string,
      }).isRequired,
    }),
  ),
};
Clients.defaultProps = {
  data: null,
};

export default Clients;
