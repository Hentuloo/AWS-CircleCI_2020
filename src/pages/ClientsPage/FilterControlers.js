import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Input from 'components/Input';
import CheckboxMultiOptions from 'components/CheckboxMultiOptions';

import sortNumbers from 'assets/icons/sortNumbers.svg';
import sortNumbersReserve from 'assets/icons/sortNumbersReserve.svg';

import { Constants } from 'config/Constants';

const Filters = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 92%;
  grid-row-gap: 10px;
  margin: 0px auto;
`;
const Title = styled.span`
  grid-column: 1/-1;
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  color: ${({ theme }) => theme.color.blue[1]};
`;
const TextInputFilter = styled(Input)`
  grid-column: 1/-1;
  width: 80%;
  margin: 0px auto;
`;
const TextWithBadge = styled.div`
  display: flex;
  flex-direction: column;
  ${({ isActive }) =>
    isActive &&
    css`
      background-color: ${({ theme }) => theme.color.blue[0]};
      color: ${({ theme }) => theme.color.white[0]};
    `}
`;
const TextWithIcon = styled.div`
  display: grid;
  height: 100%;
  grid-template-columns: auto 40px;
  padding: 0px 15px;
  align-items: center;
  ${({ isActive }) =>
    isActive &&
    css`
      background-color: ${({ theme }) => theme.color.blue[0]};
      color: ${({ theme }) => theme.color.white[0]};
    `}
`;
const SmallBadgeText = styled.small`
  font-size: ${({ theme }) => theme.fs.mini};
  letter-spacing: 0px;
  font-weight: lighter;
`;
const Icon = styled.img`
  display: block;
  height: 39px;
`;

const FilterControlers = ({
  filtersState: { sortName, sortEmail, sortAge },
  onChangeFilter,
}) => {
  return (
    <Filters>
      <TextInputFilter
        type="text"
        name="sortName"
        placeholder={Constants.pl.STATEMENTS.findClient}
        onChange={onChangeFilter}
        value={sortName}
      />
      <Title>{Constants.pl.STATEMENTS.sort.alt}</Title>
      <CheckboxMultiOptions
        options={[
          {
            value: Constants.sortTypes.default,
            name: Constants.pl.STATEMENTS.sort.default,
          },
          {
            value: Constants.sortTypes.sortEmail.desc,
            name: Constants.pl.STATEMENTS.sort.alfDesc,
          },
          {
            value: Constants.sortTypes.sortEmail.asc,
            name: Constants.pl.STATEMENTS.sort.alfAsc,
          },
        ]}
        name="sortEmail"
        defaultOption={Constants.sortTypes.default}
        onChange={onChangeFilter}
        title={Constants.pl.STATEMENTS.sort.email}
      >
        <TextWithBadge
          isActive={sortEmail !== Constants.sortTypes.default}
        >
          <span>
            {sortEmail === Constants.sortTypes.sortEmail.asc
              ? 'CBA'
              : 'ABC'}
          </span>
          <SmallBadgeText>(email)</SmallBadgeText>
        </TextWithBadge>
      </CheckboxMultiOptions>
      <CheckboxMultiOptions
        options={[
          {
            value: Constants.sortTypes.default,
            name: Constants.pl.STATEMENTS.sort.default,
          },
          {
            value: Constants.sortTypes.sortAge.desc,
            name: Constants.pl.STATEMENTS.sort.ascending,
          },
          {
            value: Constants.sortTypes.sortAge.asc,
            name: Constants.pl.STATEMENTS.sort.descending,
          },
        ]}
        name="sortAge"
        defaultOption={Constants.sortTypes.default}
        onChange={onChangeFilter}
        title={Constants.pl.STATEMENTS.sort.age}
      >
        <TextWithIcon
          isActive={sortAge !== Constants.sortTypes.default}
        >
          <span>Wiek</span>
          {sortAge === Constants.sortTypes.sortAge.desc && (
            <Icon
              src={sortNumbers}
              alt={Constants.pl.STATEMENTS.sort.descending}
            />
          )}
          {sortAge === Constants.sortTypes.sortAge.asc && (
            <Icon
              src={sortNumbersReserve}
              alt={Constants.pl.STATEMENTS.sort.ascending}
            />
          )}
        </TextWithIcon>
      </CheckboxMultiOptions>
    </Filters>
  );
};

FilterControlers.propTypes = {
  onChangeFilter: PropTypes.func.isRequired,
  filtersState: PropTypes.shape({
    sortName: PropTypes.string,
    sortEmail: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    sortAge: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  }),
};
FilterControlers.defaultProps = {
  filtersState: {
    sortName: '',
    sortEmail: '0',
    sortAge: '0',
  },
};

export default FilterControlers;
