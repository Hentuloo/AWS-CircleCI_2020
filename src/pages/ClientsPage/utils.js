import {
  isIncludeText,
  sortByAlphabet,
  sortByNumber,
} from 'config/utils';
import { Constants } from 'config/Constants';

export const filterClientsArray = (clientsArray, filterState) => {
  const { sortName, sortEmail, sortAge } = filterState;

  let clients =
    clientsArray.length &&
    clientsArray.filter(({ name }) => isIncludeText(name, sortName));
  if (sortEmail === Constants.sortTypes.sortEmail.asc) {
    clients = sortByAlphabet(clients, 'email');
  } else if (sortEmail === Constants.sortTypes.sortEmail.desc) {
    clients = sortByAlphabet(clients, 'email', true);
  }
  if (sortAge === Constants.sortTypes.sortAge.desc) {
    clients = sortByNumber(clients, 'age', true);
  } else if (sortAge === Constants.sortTypes.sortAge.asc) {
    clients = sortByNumber(clients, 'age');
  }
  return clients;
};
