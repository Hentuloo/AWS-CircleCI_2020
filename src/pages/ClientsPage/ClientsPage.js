import React, { useReducer, useMemo } from 'react';
import styled from 'styled-components';

import Clients from 'assets/clients.json';
import { Constants } from 'config/Constants';
import FilterControlers from './FilterControlers';

import ClientsList from './ClientsList';
import { filterClientsArray } from './utils';

const Wrapper = styled.main`
  ${({ theme }) => theme.mediaQuery.lg} {
    width: 90%;
    max-width: 1200px;
    margin: 0px auto;
  }
`;
const Bar = styled.aside`
  position: fixed;
  width: 100%;
  height: 70px;
  top: 0%;
  left: 0%;
  background-color: ${({ theme }) => theme.color.blue[0]};
  padding-left: 20px;
  z-index: 10;
  ${({ theme }) => theme.mediaQuery.md} {
    height: 100vh;
    width: 120px;
  }
`;
const BarTitle = styled.h1`
  font-weight: bold;
  line-height: 70px;
  text-transform: uppercase;
`;
const Content = styled.div`
  margin-top: 90px;
  ${({ theme }) => theme.mediaQuery.md} {
    margin-top: 10px;
    margin-left: 120px;
  }
`;

const App = () => {
  const { data: ClientsArray } = Clients;
  const [filterValues, setFilterValues] = useReducer(
    (state, newState) => ({ ...state, ...newState }),
    {
      sortName: '',
      sortEmail: Constants.sortTypes.default,
      sortAge: Constants.sortTypes.default,
    },
  );

  const handleChangeFilter = e => {
    const { value, name } = e.target;
    if (filterValues[name] !== value) {
      setFilterValues({ [name]: value });
    }
  };

  const FilteredClients = useMemo(
    () => filterClientsArray(ClientsArray, filterValues),
    [ClientsArray, filterValues],
  );

  return (
    <Wrapper>
      <Bar>
        <BarTitle>Klienci</BarTitle>
      </Bar>
      <Content>
        <FilterControlers
          filtersState={filterValues}
          onChangeFilter={handleChangeFilter}
        />
        <ClientsList data={FilteredClients} />
      </Content>
    </Wrapper>
  );
};

export default App;
