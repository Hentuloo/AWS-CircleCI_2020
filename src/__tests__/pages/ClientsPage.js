import React from 'react';
import { withTheme } from 'config/testUtils';
import ClientsPage from 'pages/ClientsPage/ClientsPage';
import mockClients from '__mocks__/mockyClients.json';
import { fireEvent } from '@testing-library/react';
import { Constants } from 'config/Constants';

const elementWithActiveSortText = 'onActive-text';

jest.mock('assets/clients.json', () =>
  // eslint-disable-next-line global-require
  require('__mocks__/mockyClients.json'),
);

const setup = () => {
  const el = withTheme(<ClientsPage />);
  return {
    ...el,
  };
};

describe('Page with clients list and filters controlers', () => {
  test('should display mocked clients', () => {
    const { getByText } = setup();
    const firstClientCard = getByText(mockClients.data[0].name);
    expect(firstClientCard).toBeInTheDocument();
  });

  test('should sort clients by email', () => {
    const { queryAllByTestId, queryByTitle } = setup();

    const buttonEmailDesc = queryByTitle(
      Constants.pl.STATEMENTS.sort.alfDesc,
    );
    const activeText = queryAllByTestId(elementWithActiveSortText)[0];
    let clientsCards = queryAllByTestId('user-card-component');

    expect(clientsCards[0]).toHaveTextContent(
      mockClients.data[0].name,
    );
    // second client
    expect(clientsCards[0]).not.toHaveTextContent(
      mockClients.data[1].name,
    );

    fireEvent.click(buttonEmailDesc);

    expect(activeText).toHaveTextContent(
      Constants.pl.STATEMENTS.sort.email.textDesc,
    );

    // update order of clientsCards
    clientsCards = queryAllByTestId('user-card-component');
    // check users card order
    expect(clientsCards[0]).toHaveTextContent(
      mockClients.data[1].name,
    );
    expect(clientsCards[0]).not.toHaveTextContent(
      mockClients.data[0].name,
    );
  });

  test('should sort clients by age', () => {
    const { queryAllByTestId, queryByTitle } = setup();

    const activeText = queryAllByTestId(elementWithActiveSortText)[1];
    const buttonAgeDesc = queryByTitle(
      Constants.pl.STATEMENTS.sort.descending,
    );
    let clientsCards = queryAllByTestId('user-card-component');

    expect(clientsCards[0]).toHaveTextContent(
      mockClients.data[0].age,
    );
    // second client
    expect(clientsCards[0]).not.toHaveTextContent(
      mockClients.data[1].age,
    );

    fireEvent.click(buttonAgeDesc);

    expect(activeText).toHaveTextContent(
      Constants.pl.STATEMENTS.sort.age.textDesc,
    );

    // update order of clientsCards
    clientsCards = queryAllByTestId('user-card-component');
    // check users card order
    expect(clientsCards[0]).toHaveTextContent(
      mockClients.data[1].age,
    );
    expect(clientsCards[0]).not.toHaveTextContent(
      mockClients.data[0].age,
    );
  });

  test('should filter clients by input', () => {
    const { queryByPlaceholderText, queryAllByTestId } = setup();

    const inputNode = queryByPlaceholderText(
      Constants.pl.STATEMENTS.findClient,
    );
    let clientsCards = queryAllByTestId('user-card-component');

    expect(clientsCards.length).toBe(2);

    fireEvent.change(inputNode, {
      target: { value: mockClients.data[0].name },
    });

    // check select element
    expect(inputNode).toHaveValue(mockClients.data[0].name);

    // update order of clientsCards
    clientsCards = queryAllByTestId('user-card-component');
    // check users card order
    expect(clientsCards.length).toBe(1);
  });

  test('should display statement when clientsArray is empty', () => {
    const {
      queryByPlaceholderText,
      queryAllByTestId,
      getByText,
    } = setup();

    const inputNode = queryByPlaceholderText(
      Constants.pl.STATEMENTS.findClient,
    );
    let clientsCards = queryAllByTestId('user-card-component');

    expect(clientsCards.length).toBe(2);

    fireEvent.change(inputNode, {
      target: { value: 'blah blah blah' },
    });

    // check select element
    expect(inputNode).toHaveValue('blah blah blah');

    // update order of clientsCards
    clientsCards = queryAllByTestId('user-card-component');
    // check users card order
    expect(clientsCards.length).toBe(0);

    expect(
      getByText(Constants.pl.STATEMENTS.sort.empty),
    ).toBeInTheDocument();
  });
});
