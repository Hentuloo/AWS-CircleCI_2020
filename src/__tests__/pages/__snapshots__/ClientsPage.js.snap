// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page with clients list and filters controlers mocked clients snapshot 1`] = `
Array [
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 95%;
  max-width: 450px;
  margin: 0px auto;
  border-radius: 16px;
  box-shadow: 6px 6px 8px rgba(0,0,0,0.25);
}

.c2 {
  display: grid;
  grid-template-columns: 1fr auto;
  width: 100%;
  min-height: 70px;
  padding: 5px 8px;
  border-radius: 16px 16px 0px 0px;
  background-color: rgb(166,229,238);
  margin-bottom: 15px;
}

.c3 {
  padding: 5px 0px;
}

.c4 {
  display: inline-block;
  font-weight: bold;
  font-size: 1.5em;
  margin-right: 5px;
}

.c5 {
  font-weight: bold;
  font-size: 1.0em;
  color: rgb(148,148,148);
}

.c6 {
  display: block;
  font-size: 1.2em;
  color: rgb(119,119,119);
  word-break: break-all;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c10 {
  position: absolute;
  width: 130%;
  max-height: 130%;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 50%;
}

.c8 {
  position: relative;
  border-radius: 50%;
  background-color: rgb(255,255,255);
  border: 3px solid rgb(166,229,238);
  overflow: hidden;
  margin: 0px 5px 0px 0px;
  width: 65px;
  height: 65px;
  bottom: -15px;
}

.c11 {
  position: relative;
  border-radius: 50%;
  background-color: rgb(255,255,255);
  border: 3px solid rgb(166,229,238);
  overflow: hidden;
  margin: 0px 5px 0px 0px;
  width: 30px;
  height: 30px;
  bottom: -10px;
}

.c11 .c9 {
  width: 60%;
  border-radius: 0px;
}

.c12 {
  display: grid;
  grid-template-columns: 1fr auto;
  min-height: 55px;
  padding-bottom: 10px;
}

.c13 {
  display: grid;
  grid-column-gap: 10px;
  grid-template-columns: 45px 1fr;
  grid-template-rows: repeat(3,auto);
  padding-left: 15px;
}

.c13 p {
  font-weight: bold;
}

.c14 {
  grid-row: 1/-1;
  max-width: 100%;
  max-height: 100%;
}

.c15 {
  font-size: 1.65em;
  background-color: rgb(166,229,238);
  border: none;
  color: rgb(255,255,255);
  border-radius: 8px;
  cursor: pointer;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  width: 50px;
  height: 25px;
  margin: 0px 10px 0px 0px;
}

.c15:hover {
  background-color: rgb(70,147,158);
}

.c15 img {
  max-width: 90%;
  -webkit-transform: translateY(-25%);
  -ms-transform: translateY(-25%);
  transform: translateY(-25%);
}

.c1 {
  margin: 25px auto;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {
  .c2 {
    margin-bottom: 20px;
    height: 80px;
  }
}

@media (min-width:768px) {
  .c6 {
    padding-top: 4px;
  }
}

@media (min-width:768px) {
  .c10 {
    max-width: 120%;
    max-height: 100%;
  }
}

@media (min-width:768px) {
  .c8 {
    border: 4px solid rgb(166,229,238);
  }
}

@media (min-width:768px) {
  .c8 {
    width: 80px;
    height: 80px;
    bottom: -25px;
  }
}

@media (min-width:1024px) {
  .c8 {
    bottom: -15px;
  }
}

@media (min-width:768px) {
  .c11 {
    border: 4px solid rgb(166,229,238);
  }
}

@media (min-width:768px) {
  .c11 {
    width: 35px;
    height: 35px;
    bottom: -15px;
  }
}

@media (min-width:1024px) {
  .c11 {
    bottom: -5px;
  }
}

@media (min-width:768px) {
  .c15 img {
    -webkit-transform: translateY(-45%);
    -ms-transform: translateY(-45%);
    transform: translateY(-45%);
  }
}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

<div
    class="c0 c1"
    data-testid="user-card-component"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <p
          class="c4"
        >
          Lewandowski Andrue
        </p>
        <small
          class="c5"
        >
          22
          lat
        </small>
        <span
          class="c6"
        >
          andrueLew@o2.pl
        </span>
      </div>
      <div
        class="c7"
      >
        <div
          class="sc-fzXfLW c8"
        >
          <img
            alt="avatar"
            class="c9 c10"
            src="https://cdn.pixabay.com/photo/2013/07/13/10/07/man-156584_960_720.png"
          />
        </div>
        <div
          class="sc-fzXfLW c11"
        >
          <img
            alt="Płeć"
            class="c9 c10"
            src="maleIcon.svg"
            title="Mężczyzna"
          />
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c13"
      >
        <img
          alt="Miejsce zamieszkania"
          class="c14"
          src="placeIcon.svg"
        />
        <p>
          Warsaw
        </p>
        <p>
          Miodowa
        </p>
        <p>
          18
        </p>
      </div>
      <button
        class="sc-fzXfLO c15"
        disabled=""
        title="Więcej informacji"
        type="button"
      >
        <span
          class="sr-only"
        >
          Więcej informacji
        </span>
        <img
          alt="Przycisk więcej informacji"
          src="circles.svg"
        />
      </button>
    </div>
  </div>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 95%;
  max-width: 450px;
  margin: 0px auto;
  border-radius: 16px;
  box-shadow: 6px 6px 8px rgba(0,0,0,0.25);
}

.c2 {
  display: grid;
  grid-template-columns: 1fr auto;
  width: 100%;
  min-height: 70px;
  padding: 5px 8px;
  border-radius: 16px 16px 0px 0px;
  background-color: rgb(166,229,238);
  margin-bottom: 15px;
}

.c3 {
  padding: 5px 0px;
}

.c4 {
  display: inline-block;
  font-weight: bold;
  font-size: 1.5em;
  margin-right: 5px;
}

.c5 {
  font-weight: bold;
  font-size: 1.0em;
  color: rgb(148,148,148);
}

.c6 {
  display: block;
  font-size: 1.2em;
  color: rgb(119,119,119);
  word-break: break-all;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c10 {
  position: absolute;
  width: 130%;
  max-height: 130%;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 50%;
}

.c8 {
  position: relative;
  border-radius: 50%;
  background-color: rgb(255,255,255);
  border: 3px solid rgb(166,229,238);
  overflow: hidden;
  margin: 0px 5px 0px 0px;
  width: 65px;
  height: 65px;
  bottom: -15px;
}

.c11 {
  position: relative;
  border-radius: 50%;
  background-color: rgb(255,255,255);
  border: 3px solid rgb(166,229,238);
  overflow: hidden;
  margin: 0px 5px 0px 0px;
  width: 30px;
  height: 30px;
  bottom: -10px;
}

.c11 .c9 {
  width: 60%;
  border-radius: 0px;
}

.c12 {
  display: grid;
  grid-template-columns: 1fr auto;
  min-height: 55px;
  padding-bottom: 10px;
}

.c13 {
  display: grid;
  grid-column-gap: 10px;
  grid-template-columns: 45px 1fr;
  grid-template-rows: repeat(3,auto);
  padding-left: 15px;
}

.c13 p {
  font-weight: bold;
}

.c14 {
  grid-row: 1/-1;
  max-width: 100%;
  max-height: 100%;
}

.c15 {
  font-size: 1.65em;
  background-color: rgb(166,229,238);
  border: none;
  color: rgb(255,255,255);
  border-radius: 8px;
  cursor: pointer;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  width: 50px;
  height: 25px;
  margin: 0px 10px 0px 0px;
}

.c15:hover {
  background-color: rgb(70,147,158);
}

.c15 img {
  max-width: 90%;
  -webkit-transform: translateY(-25%);
  -ms-transform: translateY(-25%);
  transform: translateY(-25%);
}

.c1 {
  margin: 25px auto;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {
  .c2 {
    margin-bottom: 20px;
    height: 80px;
  }
}

@media (min-width:768px) {
  .c6 {
    padding-top: 4px;
  }
}

@media (min-width:768px) {
  .c10 {
    max-width: 120%;
    max-height: 100%;
  }
}

@media (min-width:768px) {
  .c8 {
    border: 4px solid rgb(166,229,238);
  }
}

@media (min-width:768px) {
  .c8 {
    width: 80px;
    height: 80px;
    bottom: -25px;
  }
}

@media (min-width:1024px) {
  .c8 {
    bottom: -15px;
  }
}

@media (min-width:768px) {
  .c11 {
    border: 4px solid rgb(166,229,238);
  }
}

@media (min-width:768px) {
  .c11 {
    width: 35px;
    height: 35px;
    bottom: -15px;
  }
}

@media (min-width:1024px) {
  .c11 {
    bottom: -5px;
  }
}

@media (min-width:768px) {
  .c15 img {
    -webkit-transform: translateY(-45%);
    -ms-transform: translateY(-45%);
    transform: translateY(-45%);
  }
}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

<div
    class="c0 c1"
    data-testid="user-card-component"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <p
          class="c4"
        >
          Julian Ochota
        </p>
        <small
          class="c5"
        >
          33
          lat
        </small>
        <span
          class="c6"
        >
          julian@fmail.com
        </span>
      </div>
      <div
        class="c7"
      >
        <div
          class="sc-fzXfLW c8"
        >
          <img
            alt="avatar"
            class="c9 c10"
            src="https://cdn4.iconfinder.com/data/icons/professional-avatar-5/48/Public_relation_avatar_women_woman_profile_character_professions_2-512.png"
          />
        </div>
        <div
          class="sc-fzXfLW c11"
        >
          <img
            alt="Płeć"
            class="c9 c10"
            src="femaleIcon.svg"
            title="Kobieta"
          />
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c13"
      >
        <img
          alt="Miejsce zamieszkania"
          class="c14"
          src="placeIcon.svg"
        />
        <p>
          Gerenada
        </p>
        <p>
          call de christo
        </p>
        <p>
          23A
        </p>
      </div>
      <button
        class="sc-fzXfLO c15"
        disabled=""
        title="Więcej informacji"
        type="button"
      >
        <span
          class="sr-only"
        >
          Więcej informacji
        </span>
        <img
          alt="Przycisk więcej informacji"
          src="circles.svg"
        />
      </button>
    </div>
  </div>,
]
`;
